tests/blackbox.py

```diff
@@ -440,3 +477,412 @@ def dryrun_one(
         return self.executor(compiler_version).run_one(
             args, txn_params=txn_params, verbose=verbose
         )
+
+
+class _AutoPathDict(defaultdict):
+    def __init__(self):
+        super().__init__(lambda: _AutoPathDict())
+
+
+def as_on_complete(oc_str: str) -> OnComplete:
+    match oc_str:
+        case "no_op":
+            return OnComplete.NoOpOC
+        case "opt_in":
+            return OnComplete.OptInOC
+        case "close_out":
+            return OnComplete.CloseOutOC
+        case "clear_state":
+            return OnComplete.ClearStateOC
+        case "update_application":
+            return OnComplete.UpdateApplicationOC
+        case "delete_application":
+            return OnComplete.DeleteApplicationOC
+
+    raise ValueError(f"unrecognized {oc_str=}")
+
+
+def negate_cc(cc: CallConfig) -> CallConfig:
+    return CallConfig(3 - cc)
+
+
+class RouterSimulation:
+    """
+    TODO: the following is out of date!!!
```

Comment:
just to be clear, you mean the comments are out of date, right?

Reply:
yes. and I thought i brought them up to day. But then last evening I changed the functionality so they're out of data again. Thanks for catching!

Reply:
looks good, resolving